src/kagglehub/datasets_helpers.py

```diff
@@ -0,0 +1,81 @@
+import logging
+from http import HTTPStatus
+from typing import List
+
+from kagglehub.clients import KaggleApiV1Client
+from kagglehub.exceptions import KaggleApiHTTPError
+from kagglehub.handle import DatasetHandle
+
+logger = logging.getLogger(__name__)
+
+
+def _create_dataset(owner_slug: str, dataset_slug: str) -> None:
+    data = {"ownerSlug": owner_slug, "slug": dataset_slug, "title": dataset_slug, "isPrivate": True}
+    api_client = KaggleApiV1Client()
+    api_client.post("/datasets/create/new", data)
+    logger.info(f"Dataset '{dataset_slug}' Created.")
+
+
+def _create_dataset_instance(dataset_handle: DatasetHandle, files: List[str]) -> None:
```

Comment:
Datasets don't have `DatasetInstance` unlike models.  I'm not sure how this would be used.  This also seems quite similar to the request above, and the `owner` and `dataset` slug aren't present, so I don't know how it would figure out which dataset to append the files to?